mypy/checkexpr.py

```diff
@@ -250,6 +250,11 @@ def check_call(self, callee: Type, args: List[Node],
             self.check_argument_types(arg_types, arg_kinds, callee,
                                       formal_to_actual, context,
                                       messages=arg_messages)
+
+            ret_val_is_type_obj = is_equivalent(callee.ret_type, self.named_type('builtins.type'))
+            if callee.is_type_obj() and (len(arg_types) == 1) and ret_val_is_type_obj:
+                callee.ret_type = TypeType(arg_types[0])
```

Comment:
This seems fishy -- I wouldn't mutate callee, who knows if it's been shared elsewhere. I think the common pattern is to make a modified copy -- you can find examples of copy_modified() aplenty.